.github/workflows/main.yaml

```diff
@@ -9,8 +9,22 @@ jobs:
   gomod:
     runs-on: ubuntu-20.04
     steps:
+    - id: go-cache-paths
+      run: |
+        echo "::set-output name=go-build::$(go env GOCACHE)"
+        echo "::set-output name=go-mod::$(go env GOMODCACHE)"
     - uses: actions/checkout@v3
-    - run: make gomod
+    - name: Go Build Cache
+      uses: actions/cache@v2
+      with:
+        path: /go/pkg
+        key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+    - name: Go Mod Cache
+      uses: actions/cache@v2
+      with:
+        path: /go/pkg
+        key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}
+    - run: make go-mod-tidy
```

Comment:
Somewhat related: it seems that a few of these jobs should be consolidated into a single job with many steps. That could amortize a significant portion of the setup cost.